ddtrace/tracer/remote_config_test.go

```diff
@@ -248,6 +248,48 @@ func TestOnRemoteConfigUpdate(t *testing.T) {
 		})
 	})
 
+	t.Run("RC rule with tags", func(t *testing.T) {
+		telemetryClient := new(telemetrytest.MockClient)
+		defer telemetry.MockGlobalClient(telemetryClient)()
+		tracer, _, _, stop := startTestTracer(t, WithService("my-service"), WithEnv("my-env"))
+		defer stop()
+
+		input := remoteconfig.ProductUpdate{
+			"path": []byte(`{"lib_config": {"tracing_sampling_rate": 0.5,
+			"tracing_sampling_rules": [{
+				"service": "my-service",
+				"name": "web.request",
+				"resource": "*",
+				"provenance": "customer",
+				"sample_rate": 1.0,
+				"tags": [{"key": "tag-a", "value_glob": "tv-a??"}]
+			}]}, 
+			"service_target": {"service": "my-service", "env": "my-env"}}`),
+		}
+		applyStatus := tracer.onRemoteConfigUpdate(input)
+		require.Equal(t, state.ApplyStateAcknowledged, applyStatus["path"].State)
+
+		// A span with matching tags gets the RC rule rate
+		s := tracer.StartSpan("web.request").(*span)
+		s.SetTag("tag-a", "tv-a11")
+		s.Finish()
+		require.Equal(t, 1.0, s.Metrics[keyRulesSamplerAppliedRate])
+		require.Equal(t, samplerToDM(samplernames.RemoteUserRule), s.context.trace.propagatingTags[keyDecisionMaker])
+
+		// A span with non-matching tags gets the global rate
+		s = tracer.StartSpan("web.request").(*span)
+		s.SetTag("tag-a", "not-matching")
+		s.Resource = "abc"
```

Comment:
What's the reasoning behind setting the resource? I don't see how it would affect the outcome of the test.

Reply:
I noticed it was not set in the previous call as well. I can remove it if you want, or added it in the previous place too. Neither is a big issue, so happy to do whichever you want.

Reply:
It appears to me, there is no logical necessity to set the resource, as it doesnt affect/alter the behaviour of the test. In such case, I suggest it should be removed to decrease ambiguity for the people who will read the tests in the future

Reply:
done.